logger_cb.go

```diff
@@ -0,0 +1,97 @@
+package libbpfgo
+
+/*
+#include <bpf/libbpf.h>
+*/
+import "C"
+
+import (
+	"regexp"
+	"strings"
+
+	"github.com/aquasecurity/tracee/pkg/logger"
+)
+
+// This callback definition needs to be in a different file from where it is declared in C
+// Otherwise, multiple definition compilation error will occur
+
+//export loggerCallback
+func loggerCallback(libbpfPrintLevel int, libbpfOutput *C.char) {
+	var (
+		level    logger.Level
+		goOutput string
+	)
+
+	switch libbpfPrintLevel {
+	case C.LIBBPF_WARN:
+		level = logger.WarnLevel
+	case C.LIBBPF_INFO:
+		level = logger.InfoLevel
+	case C.LIBBPF_DEBUG:
+		level = logger.DebugLevel
+	default:
+		level = logger.ErrorLevel
+	}
+
+	goOutput = C.GoString(libbpfOutput)
+	goOutput = strings.TrimSuffix(goOutput, "\n")
+
+	if filterOutput(goOutput) {
+		return
+	}
+
+	logger.Log(level, false, goOutput)
+}
+
+func filterOutput(output string) bool {
+	var (
+		matched bool
+		err     error
+	)
+
+	// BUG: https:/github.com/aquasecurity/tracee/issues/1676
+
+	// "libbpf: Kernel error message: %s\n"
+	// 1. %s = "Esclusivity flag on"
+	// https://github.com/libbpf/libbpf/blob/68e6f83f223ebf3fbf0d94c0f4592e5e6773f0c1/src/nlattr.c#L192
```

Comment:
this long links for github don't help and polute the code, would u mind removing them ?

Reply:
> this long links for github don't help and polute the code, would u mind removing them ?

What about?

```suggestion
	// triggered by: libbpf/src/nlattr.c->libbpf_nla_dump_errormsg()

```

This would help in future research.

Reply:
yep better, thanks!